### PR TITLE
Set up Github actions CI

### PR DESCRIPTION
**Pull request by aaronlademann-wf**
Using Github actions for CI is cool, and Travis is dead.

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
<details>
<summary>(2) Vulnerable direct dependencies were detected</summary>
<li>1 vulns in <code>ini < 1.3.6</code> via <code>package-lock.json</code></li>
<li>1 vulns in <code>node-fetch < 2.6.1</code> via <code>package-lock.json</code></li>
</details>

## Action Items
* Review dependencies for available updates
* See <a href="https://splunk.workiva.net/en-US/app/search/security_vulnerable_dependencies?form.repository=react-dart">this Splunk dashboard</a> for more CVE details
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Comment by semveraudit-wf** [comment]
### Public API Changes

 No changes to the public API found for commit d8135a07e68e3c4a8b6d43984d480ee733bf0055

Showing results for d8135a07e68e3c4a8b6d43984d480ee733bf0055
> Powered by [semver-audit-service](https://github.com/Workiva/semver-audit-service). Please report any problems by [filing an issue](https://github.com/Workiva/semver-audit-service/issues).
> Reported by the dart semver audit client 2.2.0
[Browse public API](https://semver-audit.wk-dev.wdesk.org/Workiva/react-dart/d8135a07e68e3c4a8b6d43984d480ee733bf0055).

File: .github/workflows/dart_ci.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - id: install
+        name: Install dependencies
+        run: dart pub get
+
+      - name: Validate dependencies
+        run: dart pub run dependency_validator -i build_runner,build_test,build_web_compilers
+        if: always() && steps.install.outcome == 'success'
+
+      # TODO: Uncomment this, and remove the Dart 2.7.2 format step in the Workiva Build Dockerfile
+      # once we are able to bump the lower bound of SDK.
```
Comment (greglittlefield-wf) [inline .github/workflows/dart_ci.yml:34]: We don't need to keep formatting in Dart 2.7.2; before my Dockerfile PR, we had been formatting on stable (it doesn't matter to me tohugh)

File: Dockerfile [hunk 2]
```diff
@@ -74,10 +65,7 @@ WORKDIR /build/
 ADD . /build/
 
 RUN pub get
-# Run dart2js tests that fail in 2.7
-RUN pub run build_runner test --release -- --preset dart2js --tags=dart-2-7-dart2js-variadic-issues
```
Comment (greglittlefield-wf) [inline Dockerfile:78]: Can you reinstate this? I think we should continue running all the tests we can on an SDK that matches our lower bound

**Review by greglittlefield-wf — CHANGES_REQUESTED** [review]
Just one comment

File: .github/workflows/dart_ci.yml (hunk 1 above)
Comment (aaronlademann-wf) [reply .github/workflows/dart_ci.yml:34]: I think most contributors are running 2.7.x - so we should keep formatting that way IMO.

**Review by greglittlefield-wf — DISMISSED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -16,22 +16,14 @@ RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key
...
 RUN dartfmt --line-length=120 --dry-run --set-exit-if-changed .
 
-RUN pub run dependency_validator -i build_runner,build_test,build_web_compilers
-
-# TODO run tests using dart_unit_test_image in skynet, remove Chrome install
+# TODO: Remove these test runs once SDK lower bound is >=2.9.3
 RUN pub run build_runner test --release -- --preset dart2js --exclude-tags=dart-2-7-dart2js-variadic-issues
 RUN pub run build_runner test -- --preset dartdevc
 
-RUN dart ./tool/run_consumer_tests.dart --orgName Workiva --repoName over_react --testCmd "pub run dart_dev test -P dartdevc"
```
Comment (joebingham-wk) [inline Dockerfile:30]: I didn't see that this was moved elsewhere, though I might have missed it. Are we not running consumer tests as part of CI anymore?

**Review by joebingham-wk — COMMENTED** [review]
This is dope! I'm excited to switch to actions. I just had one question!

File: Dockerfile (hunk 3 above)
Comment (aaronlademann-wf) [reply Dockerfile:30]: I'd like to have them run using rmconsole. I haven't figured out why rosie isn't producing her comment to make it easier to kick off a consumer test - but they can be manually kicked off through the events interface for the time-being.

Comment (greglittlefield-wf) [reply Dockerfile:30]: Rosie comments aren't posted on OSS repos.

To see the comment, you can click the Rosie "Details" link at the bottom of the page and then click on the "Evaluation" tab.

**Review by greglittlefield-wf — APPROVED** [review]
+10

**Pull request merged**